### PR TITLE
feat(webhooks): add webhook properties for headers, method, filters and customPayload

### PR DESCRIPTION
**Pull request by kevin-mitchell**
Here I added a number of additional properties to the `Webhook` model.

I don't believe I have visibility into all of the potential `Webhook` filter `type`s, the only value I've seen in the interface is `in` and `equal`.

Similarly, I'm not sure if there are other `customPayload` types other than `text/x-handlebars-template`.

## Pull request checklist

Please check if your PR fulfills the following requirements:
- [X] Build (`npm run build`) was run locally and any changes were pushed
- [ ] Tests (`npm run test`) for the changes have been added (for bug fixes / features)
(**Note: There are a number of warnings, 44 I believe, but I don't believe this PR adds any**)
- [X] Docs (`npm run doc`) have been reviewed and added / updated if needed (for bug fixes / features)
- [X] PR title and git commit messages conform to the [conventionalcommits](https://www.conventionalcommits.org/en/v1.0.0/) specification

## Pull request type

Please check the type of change your PR introduces:
- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behaviour?

Currently unable to specify Webhook models `method`, `headers`, `customPayload`, or `filters`.

## What is the new behaviour?

- All of the above properties now exist on the typescript definition / class
- These work / are testing with the current API and allow these values to be set in the Amplience backend

## Does this introduce a breaking change?

- [ ] Yes
- [X] No

## Other information

I did not add additional tests as it didn't appear to be something currently done with `Webhook`s or in other places where types are simply being added.

Refs #82

### CODE REVIEW

**Comment by easen-amp** [comment]
Hi @kevin-mitchell,

Thank you for raising this. Looks good. 👍 

I've started to write some unit tests for your PR. Can you enable "Allow edits from maintainers" if you haven't already, so I can push these onto your branch.

Thanks.

File: src/lib/model/Webhook.ts
Comment (easen-amp) [inline src/lib/model/Webhook.ts:118]: Webhook filter arguments are 1 JSONPath argument & 1 value argument.
`arguments: [WebhookFilterJSONPathArgument, WebhookFilterValueArgument];`

Comment (easen-amp) [reply src/lib/model/Webhook.ts]: Updated.

**Review by easen-amp — APPROVED** [review]

**Comment by kevin-mitchell** [comment]
Thanks for taking the time to review and make those fixes / additions to the PR!

**Comment by easen-amp** [comment]
Our PR process requires at least 2 dev approvals and a QA approval.
I've added the relevant reviewers to this PR. 

I can't give any estimates on when this will be done, but I'm sure it won't be long.

Thanks again for contributing. 👍

**Review by simonmudd — APPROVED** [review]

**Review by Benj0s — APPROVED** [review]

**Comment by kevin-mitchell** [comment]
Thanks for all of the lovely PRs all :) 

@easen-amp any general guidance on how your PR review / release process works, e.g. when this might be merged/tagged? It be handy if I had some general/rough idea of next steps so I can do some basic planning!

Thanks again!

**Comment by easen-amp** [comment]
> Thanks for all of the lovely PRs all :)
> 
> @easen-amp any general guidance on how your PR review / release process works, e.g. when this might be merged/tagged? It be handy if I had some general/rough idea of next steps so I can do some basic planning!
> 
> Thanks again!

Sorry I missed your comment.

How it currently works it that PR's & issues are allocated to team, usually the team who wrote the feature, if they're available. Then they'll prioritise the PR/issue alongside their current work load.

So in this case, this PR has been assigned to the team who wrote the Webhooks system, my team.  

I've already reviewed the code, and I've written some unit tests for it, but we need to carry out some end-to-end testing, so just make sure it all works as expected.

We're going to take a look at doing this today/tomorrow. 👍

**Review by easen-amp — APPROVED** [review]

**Review by paulgturner — APPROVED** [review]

**Pull request merged**
